packages/core/src/components/forms/_helper-text.scss

```diff
@@ -0,0 +1,118 @@
+// Copyright 2017 Palantir Technologies, Inc. All rights reserved.
+// Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+// of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+// and https://github.com/palantir/blueprint/blob/master/PATENTS
+
+@import "../../common/variables";
+@import "../../common/mixins";
+
+/*
+Helper text
+
+Add additional helper text to help users navigate forms and complex fields.
+
+<div class="pt-callout pt-intent-primary pt-icon-info-sign">
+  Note that this feature necessitates additional DOM elements,
+  thus requiring a more complex form structure achieved with `.pt-form-group`.
+</div>
+
+- Link each label to its respective control element with a `for={#id}` attribute on the `<label>` and
+`id={#id}` on the control.
+
+- Add extra information to the label with `span.pt-text-muted`.
+
+- Add helper text to the form group to reflect user input (display intent with `.pt-form-helper-text.pt-intent-*`).
+
+- Add `.pt-inline` to `.pt-form-group` to place the label to the left of the control.
+
+- Add `.pt-large` to `.pt-form-group` to align the label when used with large Blueprint controls.
+
+Markup:
+<div class="pt-form-group">
+  <label for="example-form-group-input-a">
+    Label A
+    <span class="pt-text-muted">(optional)</span>
+  </label>
+  <div class="pt-form-content">
+    <div class="pt-input-group">
+      <span class="pt-icon pt-icon-calendar"></span>
+      <input id="example-form-group-input-a" class="pt-input" style="width: 200px;" type="text" placeholder="Input group" dir="auto" />
+    </div>
+    <div class="pt-form-helper-text">Helper text with details / user feedback</div>
+  </div>
+</div>
+<div class="pt-form-group">
+  <label for="example-form-group-input-b">
+    Label B
+    <span class="pt-text-muted">(required)</span>
+  </label>
+  <div class="pt-form-content">
+    <input id="example-form-group-input-b" class="pt-input pt-intent-danger" style="width: 300px;" type="text" dir="auto" />
+    <div class="pt-form-helper-text pt-intent-danger">Label B is required</div>
+  </div>
+</div>
+<div class="pt-form-group pt-inline pt-large">
+  <label for="example-form-group-input-c">
+    Label C
+  </label>
+  <div class="pt-form-content">
+    <div class="pt-input-group pt-large">
+      <span class="pt-icon pt-icon-calendar"></span>
+      <input id="example-form-group-input-c" class="pt-input" style="width: 200px;" type="text" placeholder="Input group" dir="auto" />
+    </div>
+    <div class="pt-form-helper-text">Helper text with details / user feedback</div>
+  </div>
+</div>
+
+Weight: 1.1
+
+Styleguide components.forms.helper-text
+*/
+
+.pt-form-group {
+  display: flex;
+  flex-direction: column;
+  margin: 0 0 $pt-grid-size;
+
+  > label {
+    margin-bottom: $pt-grid-size / 2;
+  }
+
+  > .pt-form-content > .pt-form-helper-text {
```

Comment:
It will still work, but we're specifying a pretty specific DOM structure here for the `form-group`, hence the `>`.

I can see nested form groups, but if we don't want to support, not strongly inclined.